app/javascript/components/bathroom.jsx

```diff
@@ -42,14 +38,14 @@ Bathroom.propTypes = {
   location: PropTypes.shape({
     bathroomCode: PropTypes.string,
   }),
-  loading: PropTypes.bool,
-  error: PropTypes.bool,
 };
 
 Bathroom.defaultProps = {
   location: {},
-  loading: true,
-  error: false,
 };
 
-export default withFragment(Bathroom, { location: getBathroomCode });
+export default compose(
+  renderWhileError(ErrorMessage),
+  renderWhileLoading(LoadingMessage),
```

Comment:
Why do these need to take `ErrorMessage` and `LoadingMessage`? Those placeholder components are consistent throughout the application, so I think it'd be cleaner to bake them into the respective HOCs.

Reply:
That's a good point. My first grep of the usages of `if (loading)` / `if (error)` should some alternative code paths, but those ended up being FaCC's that would be better solved by bubbling up their graphql fragments. I think defaulting the component seems reasonable.